code/__HELPERS/unsorted.dm

```diff
@@ -807,7 +808,9 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/list/turfs = new/list()
 	for(var/area/N in world)
 		if(istype(N, areatype))
-			for(var/turf/T in N) turfs += T
+			for(var/turf/T in N)
+				if(target_z == T.z || target_z == 0)
```

Comment:
Change this to `if(!target_z || target_z == T.z)`
Because 0 is the default, it's more likely to succeed the first condition, thus preventing a second check because of short circuiting behavior